src/super_gradients/training/losses/yolox_loss.py

```diff
@@ -466,3 +468,362 @@ def dynamic_k_matching(self, cost, pair_wise_ious, gt_classes, num_gt, fg_mask):
+        """
+        :param ids:                 ids of GTs, shape: [num_candidates]
+        :param ious:                pairwise IoUs, shape: [num_candidates]
+        :param dynamic_ks_bias:     multiply the resulted k to compensate the regular loss
+        """
+        assert len(ids.shape) == 1
+        assert len(ious.shape) == 1
+        assert ids.shape[0] == ious.shape[0]
+        # sort ious and ids by ious
+        ious, ious_argsort = ious.sort(descending=True)
+        ids = ids[ious_argsort]
+
+        # stable sort indices, so that ious are first sorted by id and second by value
+        ids, ids_argsort = ids.sort(stable=True)
+        ious = ious[ids_argsort]
+
+        unique_ids, ids_index_to_unique_ids_index = ids.unique_consecutive(dim=0, return_inverse=True)
+        num_unique_ids = unique_ids.shape[0]
+
+        if ids.shape[0] > 10:
+            is_in_top_10 = torch.cat(
+                (torch.ones((10,), dtype=torch.bool, device=ids.device), ids[10:] != ids[:-10]))
+        else:
+            is_in_top_10 = torch.ones_like(ids, dtype=torch.bool)
+
+        dynamic_ks = torch.zeros((num_unique_ids,), dtype=ious.dtype, device=ious.device)
+        dynamic_ks.index_put_((ids_index_to_unique_ids_index,), is_in_top_10 * ious, accumulate=True)
+        if dynamic_ks_bias is not None:
+            dynamic_ks *= dynamic_ks_bias
+        dynamic_ks = dynamic_ks.long().clamp(min=1)
+
+        all_argsort = ious_argsort[ids_argsort]
+        inverse_all_argsort = torch.zeros_like(ious_argsort)
+        inverse_all_argsort[all_argsort] = torch.arange(all_argsort.shape[0],
+                                                        dtype=all_argsort.dtype, device=all_argsort.device)
+
+        return dynamic_ks, ids_index_to_unique_ids_index[inverse_all_argsort]
+
+    def compute_cost_order(self, num_classes, candidate_gt_img_ids: torch.Tensor, candidate_gt_classes: torch.Tensor,
+                           candidate_anchor_ids: torch.Tensor, candidate_ious: torch.Tensor,
+                           cls_preds: torch.Tensor, obj_preds: torch.Tensor, strong_candidate_mask: torch.Tensor) \
+            -> torch.Tensor:
+        gt_cls_per_image = F.one_hot(candidate_gt_classes.to(torch.int64), num_classes).float()
+        with torch.cuda.amp.autocast(enabled=False):
+            cls_preds_ = (
+                    cls_preds[candidate_gt_img_ids, candidate_anchor_ids]
+                    .float().sigmoid_()
+                    * obj_preds[candidate_gt_img_ids, candidate_anchor_ids]
+                    .float().sigmoid_()
+            )
+            pair_wise_cls_cost = F.binary_cross_entropy(
+                cls_preds_.sqrt_(), gt_cls_per_image, reduction="none"
+            ).sum(-1)
+        ious_cost = -torch.log(candidate_ious + 1e-8)
+        cost = pair_wise_cls_cost + 3.0 * ious_cost + 100000.0 * strong_candidate_mask.logical_not()
```

Comment:
please make 3.0 and 100000.0 parameters of this function and of the function calling this and set them as defaults values in the compute_loss func

Reply:
I added them in the __init__ of YoloXDetectionLoss, so it can be used for the regular loss as well.
https://github.com/Deci-AI/super-gradients/blob/d8a15699e725ea7ee6a776cac2e2473f81a7265f/src/super_gradients/training/losses/yolox_loss.py#L74